p2p/net/mock/mock_test.go

```diff
@@ -273,10 +273,12 @@ func TestNetworkSetup(t *testing.T) {
 
 func TestStreams(t *testing.T) {
 
-	mn, err := FullMeshConnected(context.Background(), 3)
+	mn, err := WithNPeers(context.Background(), 3)
```

Comment:
I dont see what we lose by continuing to use `FullMeshConnected` in cases like these. Maybe just make `FullMeshConnected` be:
```go
func FullMeshConnected(...) {
  mn, err := WithNPeers(context.Background(), 3)
  if err != nil {
    nil, err
  }
  mn.LinkAll()
  mn.ConnectAll()
  return mn, nil
}
```

 i think this is different from @whyrusleeping's case.